Update 'what is a web server' (#13641)
mdn/translated-content: files/zh-cn/learn/common_questions/web_mechanics/what_is_a_web_server/index.md

---
title: 什么是 web 服务器？
slug: Learn/Common_questions/Web_mechanics/What_is_a_web_server
---

{{QuicklinksWithSubPages("Learn/Common_questions")}}

在这篇文章中我们会重温什么是 web 服务器，它们如何工作，以及为什么它们很重要。

<table>
  <tbody>
    <tr>
      <th scope="row">前提：</th>
      <td>
        你应该已经知道

>         <a href="/zh-CN/docs/Learn/Common_questions/Web_mechanics/How_the_Internet_works">互联网是怎么工作的</a

Author: ```suggestion
        <a href="/zh-CN/docs/Learn/Common_questions/Web_mechanics/How_does_the_Internet_work">互联网是怎么工作的</a
```

>         你应该已经知道
>         <a href="/zh-CN/docs/Learn/Common_questions/Web_mechanics/How_does_the_Internet_work">互联网是怎么工作的</a

Reviewer: ```suggestion
        你应该已经知道<a
          href="/zh-CN/docs/Learn/Common_questions/Web_mechanics/How_does_the_Internet_work">互联网是怎么工作的</a
```

        >，并且<a
          href="/zh-CN/docs/Learn/Common_questions/Web_mechanics/Pages_sites_servers_and_search_engines"
          >了解网页、网站、web 服务器和搜索引擎的区别是什么</a

>         >

Reviewer: ```suggestion
        >。
```

      </td>
    </tr>
    <tr>
      <th scope="row">目标：</th>
      <td>了解 web 服务器，明白它大概的工作原理</td>
    </tr>
  </tbody>
</table>

## 概述

> *网络服务器*（Web server）一词可以代指硬件或软件，或者是它们协同工作的整体。

Reviewer: ```suggestion
*web 服务器*一词可以代指硬件或软件，或者是它们协同工作的整体。
```

举个例子，要生成你在浏览器中看到的最终网页，应用服务器或许会用一个数据库中的内容填充一个 HTML 模板。像 MDN 或维基百科这样的网站有成千上万的网页。通常情况下，这类网站只由几个 HTML 模板和一个巨大的数据库组成，而不是成千上万的静态 HTML 文档。这种设置使得维护和提供内容更加容易。

## 深入探索

回顾一下：要获取一个网页，你的浏览器会向网络服务器发送一个请求，服务器会在自己的存储空间中搜索所请求的文件。找到文件后，服务器读取文件，根据需要进行处理，并将其发送给浏览器。让我们更详细地了解一下这些步骤。

### 托管文件

一个网络服务器首先需要存储这个网站的文件，也就是说所有的 HTML 文档和它们的相关资源，包括图片、CSS 样式表、JavaScript 文件、字体以及视频。

严格来说，你可以在你自己的计算机上托管所有的这些文件，但是在一个专用的 web 服务器上存储它们会方便得多，因为它

- 专用 web 服务器可用性更强（会一直启动和运行）
- 除去停机时间和系统故障，专用 web 服务器总是连接到互联网。
- 专用 web 服务器可以一直拥有一样的 IP 地址，这也称为*专有 IP 地址*（不是所有的 {{Glossary("ISP")}} 都会为家庭线提供一个固定的 IP 地址）
- 专用 web 服务器往往由第三方提供者维护

因为所有的这些原因，寻找一个优秀的托管提供者是建立你的网站的一个重要部分。比较不同公司提供的服务并选择一个适合你的需求和预算的服务（服务的价格从免费到每月上万美金不等）。你可以在[这篇文章](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/How_much_does_it_cost#专业网站机构和托管)中找到更多的细节。

一旦你设置好一个网络托管解决方案，你必须[上传你的文件到你的 web 服务器](/zh-CN/docs/Learn/Common_questions/Tools_and_setup/Upload_files_to_a_web_server)。

### 通过 HTTP 交流

其次，web 服务器提供了 {{Glossary("HTTP")}}（**H**yper**t**ext **T**ransfer **P**rotocol，超文本传输协议）支持。正如它的名字暗示，HTTP 明确提出了如何在两台计算机间传输超文本（链接的 web 文档）。

{{Glossary("Protocol", "协议")}}是一套为了在两台计算机间交流而制定的规则。HTTP 是一个文本化的（textual），无状态的（stateless）协议。

- 文本化

>   - : 所有的命令都是纯文本的（plain-text），人类可读（human-readable）。

Reviewer: ```suggestion
  - : 所有的命令都是纯文本（plain-text）且人类可读（human-readable）的。
```
